src/lambda/wizards/samInitWizard.ts

```diff
@@ -80,6 +100,171 @@ export class DefaultCreateNewSamAppWizardContext extends WizardContext implement
         return val ? (val.label as Runtime) : undefined
     }
 
+    public async promptUserForTemplate(
+        currRuntime: Runtime,
+        currTemplate?: SamTemplate
+    ): Promise<SamTemplate | undefined> {
+        const templates = getSamTemplateWizardOption(currRuntime)
+        const quickPick = picker.createQuickPick<vscode.QuickPickItem>({
+            options: {
+                ignoreFocusOut: true,
+                title: localize('AWS.samcli.initWizard.template.prompt', 'Select a SAM Application Template'),
+                value: currTemplate ? currTemplate : ''
+            },
+            buttons: [this.helpButton, vscode.QuickInputButtons.Back],
+            items: templates.toArray().map(template => ({
+                label: template,
+                alwaysShow: template === currTemplate,
+                description:
+                    template === currTemplate
+                        ? localize('AWS.wizard.selectedPreviously', 'Selected Previously')
+                        : getTemplateDescription(template)
+            }))
+        })
+
+        const choices = await picker.promptUser({
+            picker: quickPick,
+            onDidTriggerButton: (button, resolve, reject) => {
+                if (button === vscode.QuickInputButtons.Back) {
+                    resolve(undefined)
+                } else if (button === this.helpButton) {
+                    vscode.env.openExternal(vscode.Uri.parse(samInitDocUrl))
+                }
+            }
+        })
+        const val = picker.verifySinglePickerOutput(choices)
+
+        //eventBridgeStarterAppTemplate requires aws credentials
+        if (val && val.label === eventBridgeStarterAppTemplate) {
+            if (this.credentialsNotFound) {
+                await ext.awsContextCommands.onCommandLogin()
```

Comment:
This feels like a jarring UX because it pulls the user right out of the workflow, without a way back, and without knowing that it happened.

Instead, I suggest prompting the user, informing them that they need to be connected to AWS in order to select xxx.
At this point ...
* the notification could be passive and we route the user back to the same wizard step
* the notification could contain a button "Connect to AWS" which cancels the wizard and triggers onCommandLogin

I am leaning towards the first option currently, to maintain wizard flow cleanly. Users already have established ways to connect to aws.

Reply:
Went with the first option.